Framework/Muon/src/MuonGroupingCounts.cpp

```diff
@@ -0,0 +1,212 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidMuon/MuonGroupingCounts.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidHistogramData/HistogramMath.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/System.h"
+#include "MantidMuon/MuonAlgorithmHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Kernel;
+using namespace Mantid::HistogramData;
+
+namespace {
+
+bool checkPeriodInWorkspaceGroup(const int &period,
+                                 WorkspaceGroup_sptr workspace) {
+  return period <= workspace->getNumberOfEntries();
+}
+
+bool is_alnum_underscore(char c) {
```

Comment:
what is alnum?

Reply:
also could we have a longer variable name (single character variables are prone to typos)